bundles/org.openhab.binding.nibeheatpump/src/main/java/org/openhab/binding/nibeheatpump/internal/connection/SerialConnector.java

```diff
@@ -163,7 +172,6 @@ public void sendDatagram(NibeHeatPumpMessage msg) {
         public void interrupt() {
             interrupted = true;
             super.interrupt();
-            IOUtils.closeQuietly(in);
```

Comment:
Why did you remove that?

Reply:
In fact, the input stream is managed (opened and closed) by the thing handler, not by the reader thread started by the thing handler. When the conection is closed by thing handler, it first interrupts the reader thread (suing the input stream) and then closes the input stream and the serial port. This is what you see in the disconnect() method.

All bindings that I know with a reader thread are doing that like that.

Note that I applied the same change in the rme and oceanic bindings.